catboost/private/libs/algo/greedy_tensor_search.cpp

```diff
@@ -1359,6 +1365,157 @@ static TNonSymmetricTreeStructure GreedyTensorSearchLossguide(
     return currentStructure;
 }
 
+namespace {
+    struct TSubtractTrickInfo {
+        const TTrainingDataProviders* Data;
+        TVector<TCandidatesContext>* CandidatesContexts;
+        TFold* Fold;
+        TLearnContext* Ctx;
+        TQueue<TVector<TBucketStats>>* ParentsQueue;
+        double ScoreStDev;
+        int MaxBucketCount;
+        ui64 MaxSplitEnsembles;
+        ui64 StatsSize;
+        size_t MaxFeatureValueCount;
+
+        TSubtractTrickInfo(
+            const TTrainingDataProviders* data,
+            TVector<TCandidatesContext>* candidatesContexts,
+            TFold* fold,
+            TLearnContext* ctx,
+            TQueue<TVector<TBucketStats>>* parentsQueue,
+            double scoresStDev)
+            : Data(data)
+            , CandidatesContexts(candidatesContexts)
+            , Fold(fold)
+            , Ctx(ctx)
+            , ParentsQueue(parentsQueue)
+            , ScoreStDev(scoresStDev)
+        {
+            size_t nFeatures = 0;
+            MaxBucketCount = 0;
+            MaxSplitEnsembles = 0;
+            for (auto contextIdx : xrange(CandidatesContexts->size())) {
+                TCandidatesContext& candidatesContext = (*CandidatesContexts)[contextIdx];
+                nFeatures += candidatesContext.CandidateList.size();
+                for (auto candId : xrange(candidatesContext.CandidateList.size())) {
+                    MaxSplitEnsembles = std::max(MaxSplitEnsembles, candidatesContext.CandidateList[candId].Candidates.size());
```

Comment:
Please fix Android-specific build errors
To reproduce, please build as follows.
```
./ya make --build=relwithdebinfo ./catboost/private/libs/algo/ -A -DHAVE_CDUA=no --test-disable-timeout --target-platform=default-android-i686
```
```
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/catboost/private/libs/algo/greedy_tensor_search.cpp:1402:41: error: no matching function for call to 'max'
                    MaxSplitEnsembles = std::max(MaxSplitEnsembles, candidatesContext.CandidateList[candId].Candidates.size());
                                        ^~~~~~~~
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/contrib/libs/cxxsupp/libcxx/include/algorithm:2633:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('unsigned long long' vs. 'unsigned int')
max(const _Tp& __a, const _Tp& __b)
^
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/contrib/libs/cxxsupp/libcxx/include/algorithm:2644:1: note: candidate template ignored: could not match 'initializer_list<type-parameter-0-0>' against 'unsigned long long'
max(initializer_list<_Tp> __t, _Compare __comp)
^
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/contrib/libs/cxxsupp/libcxx/include/algorithm:2624:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
max(const _Tp& __a, const _Tp& __b, _Compare __comp)
^
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/contrib/libs/cxxsupp/libcxx/include/algorithm:2653:1: note: candidate function template not viable: requires single argument '__t', but 2 arguments were provided
max(initializer_list<_Tp> __t)
^
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/catboost/private/libs/algo/greedy_tensor_search.cpp:1494:12: error: moving a local object in a return statement prevents copy elision [-Werror,-Wpessimizing-move]
    return std::move(smallStats);
           ^
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/catboost/private/libs/algo/greedy_tensor_search.cpp:1494:12: note: remove std::move call here
    return std::move(smallStats);
           ^~~~~~~~~~          ~
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/catboost/private/libs/algo/greedy_tensor_search.cpp:1517:12: error: moving a local object in a return statement prevents copy elision [-Werror,-Wpessimizing-move]
    return std::move(largeStats);
           ^
/place/home/espetrov/scratch/catboost_github/pr1467/catboost/catboost/private/libs/algo/greedy_tensor_search.cpp:1517:12: note: remove std::move call here
    return std::move(largeStats);
```

Reply:
done